keps/sig-node/2413-seccomp-by-default/README.md

```diff
@@ -143,14 +143,33 @@ be extended to cover the new behavior if enabled.
 
 #### Alpha
 
-- [ ] Implement the new feature gate and kubelet configuration
-- [ ] Ensure proper tests are in place
-- [ ] Update documentation to make the feature visible
+- [x] Implement the new feature gate and kubelet configuration
+- [x] Ensure proper tests are in place
+- [x] Update documentation to make the feature visible
 
 #### Alpha to Beta Graduation
 
-- [ ] Enable the feature per default
-- [ ] No major bugs reported in the previous cycle
+- [x] Enable the feature per default
+- [x] Making the feature API aware
+- [x] No major bugs reported in the previous cycle
+
+##### API Awareness
+
+We add conversion methods to the API Server with the graduation of the feature
+to beta. The beta graduation does not mean that it's enabled by default, which
+got introduced in Kubernetes v1.24.
+
+The API Awareness allows us giving users direct feedback about the chosen seccomp
+profile as well as handling upgrade scenarios. If the feature is enabled, then
+created workloads not specifying any `securityContext.seccompProfile.type` will
+default to `RuntimeDefault`.
+
+This also means for an upgrade of a cluster, where the feature has not been
+enabled, that all empty `seccompProfile.type` fields will be converted to
+`Unconfined` to explicitly exclude already running workloads.
+
+Changing seccomp profiles of running workloads requires a restart of them, which
+incorporates with the previously described conversion.
```

Comment:
Curious, what happens under this case?
- existing cluster (from say 1.24) `seccompProfile.type` is empty.
- during upgrade (to 1.25) the cluster operator switches on this flag
- after upgrade they figure out there is a problem and switches off this flag

what would be the end result of the string in `seccompProfile.type` after steps above?

Reply:
It would be still `RuntimeDefault`, which would not solve the issue for the user. 

What we could do is to enable it only for new workloads and converting everything else to `Unconfined` if not set.

Reply:
I think we're limited to setting the field on new objects (just on pods?) at creation time

Reply:
Sounds good, I'll reflect that in the KEP. What would be a reason to only set it on pods rather than all containers (including init containers)?

Reply:
sorry, I meant just on Pod objects, rather than on other pod-spec-containing objects like Deployments, ReplicaSets, etc

Reply:
Yes, technically we only set it on the pod level, from which containers inherit.